openhands/storage/data_models/conversation_metadata.py

```diff
@@ -12,7 +13,9 @@ class ConversationTrigger(Enum):
 class ConversationMetadata:
     conversation_id: str
     github_user_id: str | None
-    selected_repository: str | None
+    # Accepting str for `selected_repository` for backward compatibility
+    # See also: https://github.com/All-Hands-AI/OpenHands/issues/7286
+    selected_repository: Repository | str | None
```

Comment:
I wonder if we should make a separate field instead of having multiple types set for `selected_repository`?

It would make the migration a little bit easier, we could just remove the deprecated field after some time

Reply:
Agree.

I've made the changes as suggested - could you please review if they align with what you had in mind?
https://github.com/All-Hands-AI/OpenHands/pull/7923/commits/bd86c4a6274f08a7bcb9015f50e63d8fe6ae11e6

One additional benefit I noticed is that separating the fields also helps simplify the type checking logic around selected_repository.